spec/integration/update_subscription_list_spec.rb

```diff
@@ -0,0 +1,59 @@
+RSpec.describe "Updating a subscriber list", type: :request do
+  let!(:subscriber_list) { create(:subscriber_list) }
+  let(:update_subscriber_lists_path) { "/subscriber-lists/#{slug}" }
+  let(:slug) { subscriber_list.slug }
+  let(:update_params) do
+    {
+      "title" => "A new title",
+    }
+  end
+
+  context "with authentication and authorisation" do
+    before do
+      login_with_internal_app
+    end
+
+    it "returns a 200" do
+      patch update_subscriber_lists_path, params: update_params.to_json, headers: json_headers
+      expect(response.status).to eq(200)
+    end
+
+    it "returns the subscriber list with updated title" do
+      patch update_subscriber_lists_path, params: update_params.to_json, headers: json_headers
+
+      expect(JSON.parse(response.body).dig("subscriber_list", "title")).to eq(update_params["title"])
+    end
+
+    it "returns 422 if no allowed permitted parameters are provided" do
+      patch update_subscriber_lists_path, params: { title: nil }.to_json, headers: json_headers
+      expect(response.status).to eq(422)
+    end
```

Comment:
How is this test passing?  In the controller you're not doing `.compact` in the params, and this is false:

```ruby
irb(main):001:0> ActionController::Parameters.new({ title: nil }).permit(:title).empty?
=> false
```

So why does this case return a 422?

Reply:
oh neat, so turns out Rails is helping.
If you try to update with unpermitted params, it is going to 422.

So the response body I'm getting is:
```
JSON.parse(response.body)
=> {"error"=>"Unprocessable Entity", "details"=>{"title"=>["can't be blank"]}}
```

And I can [drop all of this](https://github.com/alphagov/email-alert-api/pull/1704/files#diff-8eea030d9562e2c20b45db190d8639669d34b0f7017ea6e8c39fd0060de1e3efR37-R42) and get the same result

Reply:
Interestingly without that block:
```
patch update_subscriber_lists_path, params: { }.to_json, headers: json_headers
```
returns 200, which is not what we want.

So this must be ActiveRecord valdiations bubbling up from the schema requirements:
https://github.com/alphagov/email-alert-api/blob/main/db/schema.rb#L119

Reply:
Hmm, that'll work for the title, but the description is nullable, and presumably we don't want `{ "description": null }` to blank out the description.

Reply:
I think I want to test for both these scenarios, so will leave that block and add a second test.

Reply:
Amended

Reply:
I think the point is we should get two different types of 422 here, no?

1. 422 from ActiveRecord if title (or description / any future one) violates DB or ActiveRecord validation
2. 422 from our controller if we don't pass at least one permitted parameter

So I reckon if descriptions should not be nullable the thing to do would be to add validation to the subscriber list model?
Otherwise if you can have a subscriber list without a description, you should be able to patch it to null via the API, no?

Reply:
I guess we could do something like `permitted_params.empty? || permitted_params.values.all?(nil)`

Reply:
or perhaps `permitted_params.values.all?(&:blank?)` that'd get `""` too I think

Reply:
though tbh that starts to feel a bit eww, what if we want a nullable thing we can patch in future?
We're getting the controller to do validation that should be on the model?

If the problem is - what do we do with existing subscriptions that have no descriptions then I think we should scope validation to single page notificaitons only and put a migration in to add descriptions (we likely need to do that anyway?)

Reply:
Probably another question we should be asking... can publishers remove page descriptions from content?
or are they required over there?

Perhaps this one is best addressed as a part of that work?

Reply:
Description is required yes, it's the summary field in Whitehall. So every page **should** have something in the description

Reply:
If we're going to allow blanking out descriptions that's fine, but in that case we should remove the `.compact` from https://github.com/alphagov/gds-api-adapters/pull/1131/files

Reply:
Ah yeah sorry when I said migraition, i guess it could be a rake task too or something. Was thinking something that would iterate over all db values for single page notifications, likely query the content store / publishing api with the content ID and backfill all subscriber lists for pre-existing subscriber lists?

I've seen us do that with temp rake tasks and migrations on GOV.UK... I think rake tasks are better, they can best tested?